src/Features/ShippingLabelBanner.php

```diff
@@ -44,33 +39,11 @@ public function __construct() {
 	 * @return bool
 	 */
 	private function should_show_meta_box() {
-		$order = wc_get_order();
-
-		if ( ! $order ) {
-			return false;
-		}
-		// Restrict showing the metabox to supported store currencies.
-		$base_currency = get_woocommerce_currency();
-		if ( ! $this->is_supported_currency( $base_currency ) ) {
-			return false;
-		}
-
-		$base_location = wc_get_base_location();
-		if ( ! $this->is_supported_country( $base_location['country'] ) ) {
-			return false;
-		}
-
-		// At this point (no packaging data), only show if there's at least one existing and shippable product.
-		foreach ( $order->get_items() as $item ) {
-			if ( $item instanceof \WC_Order_Item_Product ) {
-				$product = $item->get_product();
-				if ( $product && $product->needs_shipping() ) {
-					return true;
-				}
-			}
+		if ( ! $this->shipping_label_banner_display_rules ) {
+			$this->shipping_label_banner_display_rules = new ShippingLabelBannerDisplayRules();
 		}
 
-		return false;
+		return $this->shipping_label_banner_display_rules->should_display_banner( JETPACK__VERSION );
```

Comment:
There's no guarantee jetpack is installed or active so `JETPACK__VERSION` may be undefined. We should stop passing it as a parameter instead do a `defined` check in `should_display_banner()` and get it if it exists.

Reply:
I need to pass the jetpack version here as a parameter because it's defined as a constant and it cannot be mocked. So I needed to decouple it from the `ShippingLabelBannerDisplayRules` class. But you're right this could fail... I'll add the check in this method instead.